File: app/src/main/java/com/carpinelli/testeandroid/ui/invest/DownInfoAdapter.java
package com.carpinelli.testeandroid.ui.invest;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.carpinelli.testeandroid.R;
import com.carpinelli.testeandroid.model.DownInfo;
import com.carpinelli.testeandroid.model.Info;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

public class DownInfoAdapter extends RecyclerView.Adapter<DownInfoAdapter.InfoViewHolder> {

    private Context context;
    private List<DownInfo> infoList;
    private DownInfo info;

    public DownInfoAdapter(Context context, List<DownInfo> infoList) {
        this.context = context;
        this.infoList = infoList;
    }


    @NonNull
    @Override
    public InfoViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.item_down_info, parent, false);
        InfoViewHolder holder = new InfoViewHolder(view);
        return holder;
    }

    @Override
    public void onBindViewHolder(@NonNull InfoViewHolder holder, int position) {
        info = infoList.get(position);

        holder.tvName.setText(info.getName());

    }

    @Override
    public int getItemCount() {
        return this.infoList != null ? infoList.size() : 0;
    }


    public static class InfoViewHolder extends RecyclerView.ViewHolder {

        @BindView(R.id.tvName)
        TextView tvName;

        public InfoViewHolder(View itemView) {
            super(itemView);

            ButterKnife.bind(this, itemView);

        }
    }


}

File: SantanderInvestimento/app/src/main/java/br/com/santander/santanderinvestimento/investiment/adapter/DownInfoAdapter.kt
package br.com.santander.santanderinvestimento.investiment.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import br.com.santander.santanderinvestimento.R
import br.com.santander.santanderinvestimento.investiment.domain.entity.Info
import kotlinx.android.synthetic.main.item_view_down_info.view.*

class DownInfoAdapter(listItems: MutableList<Info>, val clickListener: (String) -> Unit) : RecyclerView.Adapter<DownInfoAdapter.DownInfoViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): DownInfoViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_view_down_info, parent, false)
        return DownInfoViewHolder(view)
    }

    private val infoListItems: MutableList<Info> = listItems


    override fun getItemCount() = infoListItems.size

    override fun onBindViewHolder(holder: DownInfoViewHolder, position: Int) {
        val note = infoListItems[position]
        holder.bindView(note)
    }

    inner class DownInfoViewHolder(view: View) : RecyclerView.ViewHolder(view) {

        fun bindView(item: Info) {
            setIsRecyclable(false)
            itemView.txName.text = item.name
            itemView.clDownload.setOnClickListener { clickListener("") }
        }
    }
}